### DISCUSSION TITLE
Q&A from Noa3

### DISCUSSION THREAD

**Discussion by Noa3** [Noa3/Cowtastic-Open-Cafe#1] [Q&A]
Do you have any questions about the Project?

**Comment by midiconch** [comment]
![image](https://github.com/user-attachments/assets/646fa1a1-cef7-4600-a7b8-da646c5dde44)

Hey Noa3, When I pulled down the project I'm getting a lot of missing script references. Are these due to asset store scripts being removed? If not, what should be added to those objects?

**Reply by midiconch** [reply]
After looking into this further, it looks like these are visual effects on these buttons. Maybe sparkles when you click them or something? The functionality of these buttons is still intact.

From what I can tell, everything works, except for some reason the seed box is disabled. In both runtime and editor, the seed box never appears for any game mode. I wanted to mention this to you in case this isn't what you would expect, maybe indicating my environment is different or something.

On my fork, I was able to fix this by changing around the OnValueChanged for the game mode toggles

![image](https://github.com/user-attachments/assets/949fdc37-4fd9-4a66-802c-4ba83ecc53dc)

Before, this value would do something like Canvas.enabled. Meanwhile, the game object itself was inactive.

No big deal, I just thought it was odd that the Seed Text Box seemed to have been disabled basically.

I think I can see sorta what's wrong with the seeding. I'll get to that soon.

**Reply by Noa3** [reply]
these gameobjects are made as Preperated effects for the "Feel" package from the Assetstore. they play some sounds and Postprocessing effects on different events.

**Reply by Noa3** [reply]
The seedbox got deactivaded by me because it didnt work like i intended, to make a determatistic gameplay. it will probably be a huge chunk of work to get it really working.

**Reply by midiconch** [reply]
Awesome, thanks for the reply. Good to know my version of the project isn't broken somehow.

**Reply by midiconch** [reply]
Currently the issue is that something calls getrandom during startup, so the ranseed value gets set. Since it is then set, it never checks playerprefs to see if they've requested a specific seed (because it only calls getRandomSeed when ranSeed == 0).

Also, when you start the game, it appears to have forgotten the previously entered seed. This is because SetSeed.cs calls SeedSet in its start(), passing in a random value, which then overwrites the playerprefs value.

No worries, I'll fix it while I'm implementing the separate seeds per mechanic.

**Reply by midiconch** [reply]
In other words, current behavior:

1. Only check playerprefs for seed at startup
2. Just before doing that, set playerprefs to random seed
